apps/video/test/marionette/lib/video.js

```diff
@@ -147,6 +216,23 @@ Video.prototype = {
    return this.client.findElements(Video.Selector.thumbnail);
   },
 
+  get thumbnails() {
+    return this.client.findElements(Video.Selector.thumbnail);
+  },
+
+  waitForThumbnails: function(num) {
+    var thumbnails = this.thumbnails;
+
+    this.client.helper.waitFor(function() {
+      return thumbnails.length === num;
+    });
```

Comment:
The function is getting the thumbnails element. The element may or may not have been fully constructed when we first get the reference to it. The metadata loading process will add list items to the element so 'thumbnails.length' will change if the metadata loading process is still active after findElement is invoked.